cmd/jujud/agent/machine/manifolds.go

```diff
@@ -262,16 +262,11 @@ func Manifolds(config ManifoldsConfig) dependency.Manifolds {
 	// connectFilter exists:
 	//  1) to let us retry api connections immediately on password change,
 	//     rather than causing the dependency engine to wait for a while;
-	//  2) to ensure that certain connection failures correctly trigger
-	//     complete agent removal. (It's not safe to let any agent other
-	//     than the machine mess around with SetCanUninstall).
+	//  2) to decide how to deal with fatal, non-recoverable errors
+	//     e.g apicaller.ErrConnectImpossible.
 	connectFilter := func(err error) error {
 		cause := errors.Cause(err)
 		if cause == apicaller.ErrConnectImpossible {
-			err2 := coreagent.SetCanUninstall(config.Agent)
-			if err2 != nil {
-				return errors.Trace(err2)
-			}
```

Comment:
One discussion we had is potentially having us remove the Systemd file (so they agent doesn't continually start, fail to login, then die and get restarted by systemd), but *not* remove all of the /var/lib/juju/* content. (Thus in the event of triggering this accidentally, you can still recover)
That said, this is a more conservative option (just don't uninstall anything), which might be a good first step.